src/gt4py/next/errors/excepthook.py

```diff
@@ -0,0 +1,94 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2023, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+import os
+import sys
+from typing import Callable, Optional
+
+import importlib_metadata
+
+from . import exceptions, formatting
+
+
+def _get_developer_mode_python() -> bool:
+    """Guess if the Python environment is used to develop gt4py."""
+    # Import gt4py and use its __name__ because hard-coding "gt4py" would fail
+    # silently if the module's name changes for whatever reason.
```

Comment:
Yeah, what you're saying is probably right. The idea here is to avoid simply doing `s.startswith("gt4py")` because you would never realize it's wrong if the package name changed. If you `import gt4py`, that must be kept in sync with the real package name or else you'll crash with an import error, and the `__name__` is just to keep the string in sync with the import.